scipy/stats/tests/test_stats.py

```diff
@@ -5681,6 +5688,107 @@ def test_weights_masked_1d_array(self):
         check_equal_gmean(a, desired, weights=weights, rtol=1e-5)
 
 
+class TestPowMean:
+
+    def pmean_reference(a, p):
+        return (np.sum(a**p) / a.size)**(1/p)
+
+    def wpmean_reference(a, p, weights):
+        return (np.sum(weights * a**p) / np.sum(weights))**(1/p)
+
+    def test_1d_list(self):
+        #  Test a 1d list
+        a, p = [10, 20, 30, 40, 50, 60, 70, 80, 90, 100], 3.5
+        desired = TestPowMean.pmean_reference(np.array(a), p)
+        check_equal_pmean(a, p, desired)
+
+        a, p = [1, 2, 3, 4], 2
+        desired = np.sqrt((1**2 + 2**2 + 3**2 + 4**2) / 4)
+        check_equal_pmean(a, p, desired)
+
+    def test_1d_array(self):
+        #  Test a 1d array
+        a, p = np.array([10, 20, 30, 40, 50, 60, 70, 80, 90, 100]), -2.5
+        desired = TestPowMean.pmean_reference(a, p)
+        check_equal_pmean(a, p, desired)
+
+    def test_1d_array_with_zero(self):
+        a, p = np.array([1, 0]), -1
+        desired = 0.0
+        assert_equal(stats.pmean(a, p), desired)
+
+    def test_1d_array_with_negative_value(self):
+        a, p = np.array([1, 0, -1]), 1.23
+        assert_raises(ValueError, stats.pmean, a, p)
+
+    # Note the next tests use axis=None as default, not axis=0
+    def test_2d_list(self):
+        #  Test a 2d list
+        a, p = [[10, 20, 30, 40], [50, 60, 70, 80], [90, 100, 110, 120]], -0.5
+        desired = TestPowMean.pmean_reference(np.array(a), p)
+        check_equal_pmean(a, p, desired)
+
+    def test_2d_array(self):
+        #  Test a 2d array
+        a, p = [[10, 20, 30, 40], [50, 60, 70, 80], [90, 100, 110, 120]], -0.5
+        desired = TestPowMean.pmean_reference(np.array(a), p)
+        check_equal_pmean(np.array(a), p, desired)
+
+    def test_2d_axis0(self):
+        #  Test a 2d list with axis=0
+        a, p = [[10, 20, 30, 40], [50, 60, 70, 80], [90, 100, 110, 120]], -0.5
+        desired = [TestPowMean.pmean_reference(np.array(
+            [a[i][j] for i in range(len(a))]), p) for j in range(len(a[0]))]
```

Comment:
Such formatting has a low line count but is confusing, please consider updating to something like the following here and bellow (if not parametrised OC)
```suggestion
        desired = [
            TestPowMean.pmean_reference(
                np.array([a[i][j] for i in range(len(a))]), p
            )
            for j in range(len(a[0]))
        ]
```

Not a fan or `range(len(...))` as well. We removed most of them in the codebase not so long ago.